tests/test_compose.py

```diff
@@ -294,3 +295,25 @@ def test_initialization_root_module(monkeypatch: Any) -> None:
     monkeypatch.chdir("tests/test_apps/init_in_app_without_module")
     subprocess.check_call([sys.executable, "main.py"])
     subprocess.check_call([sys.executable, "-m", "main"])
+
+
+@mark.parametrize(
+    ("overrides", "expected"),
+    [
+        param([], {"map": {}}, id="default"),
+        param(["map.foo=bar"], {"map": {"foo": "bar"}}, id="add_no_plus"),
+        param(["+map.foo=bar"], {"map": {"foo": "bar"}}, id="add_no_plus"),
+    ],
+)
+def test_adding_to_sc_dict(
+    hydra_restore_singletons: Any, overrides: List[str], expected: Any
+) -> None:
+    @dataclass
+    class Config:
+        map: Dict[str, str] = field(default_factory=dict)
+
+    ConfigStore.instance().store(name="config", node=Config)
+
+    with initialize():
+        cfg = compose(config_name="config", overrides=overrides)
+        assert cfg == expected
```

Comment:
@jieru-hu, this is a new a more lightweight way to test config composition than using a test app.